cli/hooks/webpack.js

```diff
@@ -0,0 +1,177 @@
+	});
+	return webpackLogger;
+};
+
+exports.id = 'ti.webpack';
+exports.init = (logger, config, cli) => {
+	const isAppcCli = typeof process.env.APPC_ENV !== 'undefined';
+	let commandName;
+	let isWebpackEnabled = false;
+	let projectType;
+	let client;
+
+	cli.on('cli:command-loaded', (hookData) => {
+		const command = hookData.command;
+		commandName = command.name;
+	});
+
+	cli.on('cli:post-validate', () => {
+		projectType = getWebpackProjectType(cli.argv['project-dir']);
+		if (projectType === null) {
+			return;
+		}
+		isWebpackEnabled = true;
+		process.env.TI_USE_WEBPACK = true;
+
+		if (commandName !== 'build') {
+			return;
+		}
+
+		let appcdRootPath;
+		if (isAppcCli) {
+			// we were invoked by appc-cli, load bundled daemon client.
+			appcdRootPath = resolveModuleRoot('appcd');
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find Appcelerator Daemon inside the appc-cli. Please make sure to use a recent appc-cli version. You can install/select it with  "appc use latest".');
+			}
+		} else {
+			// plain ti-cli, load daemon from global modules
+			const globalModulesPath = execSync('npm root -g').toString().replace(/\s+$/, '');
+			appcdRootPath = resolveModuleRoot('appcd', [ globalModulesPath ]);
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find global Appcelerator Daemon install. You can install it with "npm i appcd -g".');
+			}
+		}
+		ensureAppcdVersion(appcdRootPath, MIN_APPCD_VERSION, isAppcCli);
+		const AppcdClient = resolveAppcdClient(appcdRootPath);
+		const baseClient = new AppcdClient();
+		if (isAppcCli) {
+			baseClient.appcd = path.resolve(appcdRootPath, '..', '.bin', 'appcd');
+			if (process.platform === 'win32') {
+				baseClient.appcd += '.cmd';
+			}
+		}
+		client = appcd(baseClient);
+	});
+
+	cli.on('build.pre.compile', {
+		priority: 800,
+		post(builder, callback) {
+			if (!isWebpackEnabled) {
+				return callback();
+			}
+
+			builder.useWebpack = true;
+
+			const badgedLogger = createWebpackLogger(logger);
+			const webpackService = new WebpackService({
+				client,
+				logger: badgedLogger,
+				cli,
+				builder,
+				projectType
+			});
+			Promise.resolve().then(async () => {
+				await webpackService.build();
+				return callback();
+			}).catch(e => {
+				if (e.status === 404) {
+					badgedLogger.info('Daemon was unable to find the Webpack plugin. To continue you need to:');
+					badgedLogger.info('');
+					const installCommand = 'npm i -g @appcd/webpack-plugin';
+					badgedLogger.info(`- Install it with ${installCommand.cyan}`);
+					const restartCommand = `${isAppcCli ? 'appc ' : ''}appcd restart`;
+					badgedLogger.info(`- Restart the daemon with ${restartCommand.cyan}`);
+					badgedLogger.info('');
+				}
+
+				if (cli.argv.platform === 'ios') {
+					// The iOS build shows the error message only, log the stack manually
+					// to make sure it get printed as well
+					logger.error(e.stack);
+				}
+
+				callback(e);
```

Comment:
<!--
  0 failure: 
  1 warning:  cli/hooks/webpack...
  
  
  DangerID: danger-id-Danger;
  File: cli/hooks/webpack.js;
  Line: 115;
-->

- :warning: cli/hooks/webpack.js line 115 – Avoid calling back inside of a promise. (promise/no-callback-in-promise)